android/app/src/main/java/com/chorestory/fragment/ParentCreateFragment.java

```diff
@@ -0,0 +1,252 @@
+            @Override
+            public void onClick(View v) {
+                disableButtons();
+                final Calendar c = Calendar.getInstance();
+                int hour = c.get(Calendar.HOUR_OF_DAY);
+                int minute = c.get(Calendar.MINUTE);
+
+                TimePickerDialog timePickerDialog = new TimePickerDialog(getActivity(),
+                        new TimePickerDialog.OnTimeSetListener() {
+                            @Override
+                            public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+                                // TODO: consider using AM/PM instead
+                                String time = String.format("%02d:%02d", hourOfDay, minute);
+                                mHour = hourOfDay;
+                                mMinute = minute;
+                                timeTextView.setText(time);
+                            }
+                        },
+                        hour,
+                        minute,
+                        DateFormat.is24HourFormat(getActivity()));
+                timePickerDialog.show();
+                enableButtons();
+            }
+        });
+
+        ArrayAdapter<CharSequence> recurrenceSpinnerAdapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.recurrence_array, R.layout.spinner_item);
+
+        recurrenceSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        recurrenceSpinner.setAdapter(recurrenceSpinnerAdapter);
+
+        recurrenceSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                recurrenceType = (String) parent.getItemAtPosition(position);
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+                recurrenceType = null;
+            }
+        });
+
+        mandatoryCheckBox.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mandatory = mandatoryCheckBox.isChecked();
+            }
+        });
+
+        createQuestFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                disableButtons();
+                if (questType == null ||
+                        exp < 0 ||
+                        mYear < 0 ||
+                        mMonth < 0 ||
+                        mDay < 0 ||
+                        mHour < 0 ||
+                        mMinute < 0 ||
+                        recurrenceType == null ||
+                        questType.equals(getString(R.string.select_a_quest))) {
+                    // TODO: think of message; make it a string resource
+                    Toaster.showToast(getActivity(), "Invalid Quest");
```

Comment:
We're probably going to have to hook in more error checking and logic for the voice aspect in this onClick.
Can we pre-emptively move these checks into a `QuestCreationHandler` and add more informative errors (like telling the user which field is incorrect)